### PR TITLE
Hide Highlights of plugin if not in focused

### PR DESCRIPTION
**Pull request by LianaHus**

### CODE REVIEW

File: apps/remix-ide/src/app/editor/SourceHighlighters.js [hunk 1]
```diff
@@ -28,13 +25,40 @@ class SourceHighlighters {
     }
   }
 
+  highlightAllFrom (from) {
```
Comment (yann300) [inline apps/remix-ide/src/app/editor/SourceHighlighters.js:28]: could you put some doc to this function?

File: apps/remix-ide/src/app/editor/SourceHighlighters.js [hunk 2]
```diff
@@ -28,13 +25,40 @@ class SourceHighlighters {
...
+    try {
+      if (!this.highlighters[from]) return
+      const sourceHighlight = new SourceHighlighter()
```
Comment (yann300) [inline apps/remix-ide/src/app/editor/SourceHighlighters.js:31]: From what I see you don't need to re-instanciate each time a `sourceHighlight = new SourceHighlighter()` but only if it doesn't already exist

Comment (LianaHus) [reply apps/remix-ide/src/app/editor/SourceHighlighters.js:31]: they keep an ad for each marker and if I don't do it the new marker has different id

File: apps/remix-ide/src/app/editor/sourceHighlighter.js [hunk 3]
```diff
@@ -39,6 +39,7 @@ class SourceHighlighter {
     this.source = null
     if (lineColumnPos) {
       this.source = filePath
+      if (!this.source) this.source = this._deps.fileManager.currentFile()
```
Comment (yann300) [inline apps/remix-ide/src/app/editor/sourceHighlighter.js:42]: Why do you need to do that? i'd say if `filePath` is not defined the function can't anyway process and should return an error or just return.

File: apps/remix-ide/src/app/editor/editor.js [hunk 4]
```diff
@@ -189,14 +189,17 @@ class Editor extends Plugin {
...
       this.event.trigger('sessionSwitched', [])
-
       this.editor.getSession().on('change', () => {
         this._onChange()
         this.event.trigger('contentChanged', [])
       })
     })
   }
 
+  onActivation () {
```
Comment (yann300) [inline apps/remix-ide/src/app/editor/editor.js:199]: might need to do something like `onDeactivation` : `this.off('sidePanel', 'focusChanged')` .
Please check the syntax.

**Comment by yann300** [comment]
That should probably need a little e2e test. 
At least to test that highlight is removed when switched to another view

File: apps/remix-ide/src/app/editor/SourceHighlighters.js (hunk 2 above)
Comment (yann300) [reply apps/remix-ide/src/app/editor/SourceHighlighters.js:31]: > ad ?

File: apps/remix-ide/src/app/editor/editor.js (hunk 4 above)
Comment (LianaHus) [reply apps/remix-ide/src/app/editor/editor.js:199]: right

File: apps/remix-ide/src/app/editor/SourceHighlighters.js (hunk 2 above)
Comment (LianaHus) [reply apps/remix-ide/src/app/editor/SourceHighlighters.js:31]: sorry. "an id"

File: apps/remix-ide/src/app/editor/SourceHighlighters.js [hunk 5]
```diff
@@ -28,13 +25,40 @@ class SourceHighlighters {
...
+      for (const index in this.highlighters[from]) {
+        sourceHighlight.currentSourceLocationFromfileName(
+          this.highlighters[from][index].position,
+          this.highlighters[from][index].filePath,
```
Comment (yann300) [inline apps/remix-ide/src/app/editor/SourceHighlighters.js:35]: in which case do you have `filePath` here ? I didn't see `filePath` but `source` is present. Also `hexColor` seems not

Comment (yann300) [reply apps/remix-ide/src/app/editor/SourceHighlighters.js:35]: ^ might be the reason why you have `filePath` undefined and that you needed to to `if (!this.source) this.source = this._deps.fileManager.currentFile()`

File: apps/remix-ide/src/app/editor/contextualListener.js [hunk 6]
```diff
@@ -123,14 +123,15 @@ class ContextualListener extends Plugin {
   }
 
   _highlightInternal (position, node) {
+    if (node.nodeType == 'Block') return
```
Comment (ioedeveloper) [inline apps/remix-ide/src/app/editor/contextualListener.js:126]: Can you change this to strict equality check `===` not `==`

**Review by ioedeveloper — CHANGES_REQUESTED** [review]

File: apps/remix-ide/src/app/editor/contextualListener.js (hunk 6 above)
Comment (LianaHus) [reply apps/remix-ide/src/app/editor/contextualListener.js:126]: :) of course, thanks. you are checking the standart manually :)

**Review by ioedeveloper — APPROVED** [review]

**Pull request merged**